test/integration/targets/ansible-test-sanity-validate-modules/ansible_collections/ns/col/plugins/modules/semantic_markup.py

```diff
@@ -75,6 +75,23 @@
   a10:
     description: O(foo.bar=1).
     type: str
+
+  a11:
+    description: Something with suboptions.
+    type: dict
+    suboptions:
+      b1:
+        description:
+          - V(C\(foo\)).
+          - RV(bam).
+          - P(foo.bar#baz).
+          - P(foo.bar.baz).
+          - P(foo.bar.baz#woof).
+          - E(foo\(bar).
+          - O(bar).
+          - O(bar=bam).
+          - O(foo.bar=1).
+        type: str
```

Comment:
I looked over the test coverage for the semantic markup and it seems there are a few cases not yet covered:

* format_error
* format_horizontal_line
* format_link
* format_module
* format_rst_ref

If it's not too much effort, it would be nice to have coverage for these as well.

Reply:
These are covered by be826d315e1588e4a7ef5db368c3eab1f3be77bc. That output also shows changed behavior in case of parsing errors: before this PR, `M(broken markup)` would render as `[broken markup]`; now it renders as `[[ERROR while parsing: While parsing "M(broken markup)" at index 418: Module name "broken markup" is not a FQCN]]`.

Reply:
Almost all of these cases should have already been covered by test/units/cli/test_doc.py though? (Except `format_error`.)

Reply:
Ah, I didn't even look at the unit test coverage. Just the `ansible-doc` and `ansible-test-sanity-validate-modules` integration tests.